addons/l10n_eg_edi_eta/models/account_edi_format.py

```diff
@@ -0,0 +1,426 @@
+
+    def _post_invoice_edi(self, invoices):
+
+        # ---------------------------------------------------------------------
+
+        # OVERRIDE
+        if self.code != 'eg_eta':
+            return super()._post_invoice_edi(invoices)
+        invoice = invoices  # Batching is disabled for this EDI.
+
+        if invoice.l10n_eg_submission_id:
+            return self._l10n_eg_get_einvoice_status(invoice)
+
+        if not invoice.l10n_eg_eta_invoice_id:
+            return {invoice: {
+                'error':  _("An error occured in created the ETA invoice, please retry signing"),
+                'blocking_level': 'error'
+            }
+            }
+        invoice_json = json.loads(
+            invoice.l10n_eg_eta_invoice_id.raw)['request']
+        if not invoice_json.get('signatures'):
+            return {invoice: {
+                'error':  _("Please make sure the invoice is signed"),
+                'blocking_level': 'info'
+            }
+            }
+        return self._l10n_eg_edi_post_invoice_web_service(invoice, invoice_json)
+
+    def _l10n_eg_edi_post_invoice_web_service(self, invoice, invoice_json):
+        token = self._l10n_eg_get_einvoice_token(invoice)
+        api_domain = self._l10n_eg_get_eta_api_domain()
+        if isinstance(token, dict) and token.get('error', False):
+            return {invoice: token}
+        request_url = '%s/api/v1.0/documentsubmissions' % api_domain
+
+        request_payload = {
+            'documents': [invoice_json]
+        }
+
+        data = json.dumps(request_payload, ensure_ascii=False, indent=4).encode('utf-8')
+
+        headers = {'Content-Type': 'application/json',
+                   'Authorization': 'Bearer %s' % token}
+        try:
+            request_response = requests.post(
+                request_url, data=data, headers=headers, timeout=(5, 10))
+            submission_data = request_response.json()
+        except (ValueError, requests.exceptions.ConnectionError, requests.exceptions.MissingSchema, requests.exceptions.Timeout, requests.exceptions.HTTPError) as ex:
+            return {invoice: {'error': str(ex), 'blocking_level': 'warning'}}
+
+        if submission_data:
+            if submission_data.get('error'):
+                return {invoice: {
+                    'error': submission_data.get('error'),
+                    'blocking_level': 'error'
+                }}
+            if submission_data.get('rejectedDocuments', False) and isinstance(submission_data.get('rejectedDocuments'), list):
+                return {invoice: {
+                    'error': str(submission_data.get('rejectedDocuments')[0].get('error')),
+                    'blocking_level': 'error'
+                }}
+            if (submission_data.get('submissionId') is not None
+                    and submission_data.get('acceptedDocuments')):
+                eta_invoice_json = json.loads(invoice.l10n_eg_eta_invoice_id.raw)
+                eta_invoice_json['response'] = {
```

Comment:
As we do not support batching, I suppose we could just put submission_data entirely?